cmd/kueuectl/app/create/create_clusterqueue.go

```diff
@@ -0,0 +1,409 @@
+	cmd.Flags().StringVar(&o.UserSpecifiedQueueingStrategy, queuingStrategy, "",
+		"The queueing strategy of the workloads across the queues in this ClusterQueue.")
+	cmd.Flags().StringToStringVar(&o.UserSpecifiedNamespaceSelector, namespaceSelector, map[string]string{},
+		"Defines which namespaces are allowed to submit workloads to this clusterQueue.")
+	cmd.Flags().StringVar(&o.UserSpecifiedReclaimWithinCohort, reclaimWithinCohort, "",
+		"Determines whether a pending Workload can preempt Workloads from other ClusterQueues in the cohort that are using more than their nominal quota.")
+	cmd.Flags().StringVar(&o.UserSpecifiedPreemptionWithinClusterQueue, preemptionWithinClusterQueue, "",
+		"Determines whether a pending Workload that doesn't fit within the nominal quota for its ClusterQueue, can preempt active Workloads in the ClusterQueue.")
+	cmd.Flags().StringSliceVar(&o.UserSpecifiedNominalQuota, nominalQuota, []string{},
+		"The quantity of this resource that is available for Workloads admitted by this ClusterQueue at a point in time.")
+	cmd.Flags().StringSliceVar(&o.UserSpecifiedBorrowingLimit, borrowingLimit, []string{},
+		"The maximum amount of quota for the [flavor, resource] combination that this ClusterQueue is allowed to borrow from the unused quota of other ClusterQueues in the same cohort.")
+	cmd.Flags().StringSliceVar(&o.UserSpecifiedLendingLimit, lendingLimit, []string{},
+		"The maximum amount of unused quota for the [flavor, resource] combination that this ClusterQueue can lend to other ClusterQueues in the same cohort.")
+
+	return cmd
+}
+
+// Complete completes all the required options
+func (o *ClusterQueueOptions) Complete(clientGetter genericclioptions.RESTClientGetter, cmd *cobra.Command, args []string) error {
+	o.Name = args[0]
+
+	if cmd.Flags().Changed(queuingStrategy) {
+		o.QueueingStrategy = v1beta1.QueueingStrategy(o.UserSpecifiedQueueingStrategy)
+	}
+
+	if cmd.Flags().Changed(namespaceSelector) {
+		o.NamespaceSelector = metav1.LabelSelector{
+			MatchLabels: o.UserSpecifiedNamespaceSelector,
+		}
+	}
+
+	if cmd.Flags().Changed(reclaimWithinCohort) {
+		o.ReclaimWithinCohort = v1beta1.PreemptionPolicy(o.UserSpecifiedReclaimWithinCohort)
+	}
+
+	if cmd.Flags().Changed(preemptionWithinClusterQueue) {
+		o.PreemptionWithinClusterQueue = v1beta1.PreemptionPolicy(o.UserSpecifiedPreemptionWithinClusterQueue)
+	}
+
+	if cmd.Flags().Changed(nominalQuota) || cmd.Flags().Changed(borrowingLimit) || cmd.Flags().Changed(lendingLimit) {
+		nominalQuotas := parseUserSpecifiedResourceQuotas(o.UserSpecifiedNominalQuota, nominalQuota)
+		borrowingLimits := parseUserSpecifiedResourceQuotas(o.UserSpecifiedBorrowingLimit, borrowingLimit)
+		lendingLimits := parseUserSpecifiedResourceQuotas(o.UserSpecifiedLendingLimit, lendingLimit)
+
+		resourceGroups := mergeResourcesByFlavor(slices.Concat(nominalQuotas, borrowingLimits, lendingLimits))
+		o.ResourceGroups = mergeFlavorsByCoveredResources(resourceGroups)
```

Comment:
lets make this a unit testable function say

```go
func parseResourceGroups(nominalQuotas , borrowingLimits, lendingLimits []string) ([]v1beta1.ResourceGroup, error)
```

It should make sure that the resource groups are consistent across flavors and quotas, specifying only brorrowingLimits for one flavor should be possible, specifying  lendingLimits  without  borrowing limits should be posible.